### PR TITLE
ARCO-187: change Cumulative Fees Validation behavior

### PR DESCRIPTION
**Pull request by arkadiuszos4chain**
## Description of Changes

* `Cumulative Fees Validation` behavior changed: validation doesn't check unmined ancestors chain length anymore
* new `Strict Cumulative Fees Validation` - works as CFV with ancestors chain lenght validation

## Testing Procedure

- [x] I have added new unit tests
- [x] All tests pass locally
- [x] I have tested manually in my local environment

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have updated `CHANGELOG.md` with my changes

### CODE REVIEW

File: internal/validator/default/default_validator_test.go [hunk 1]
```diff
@@ -488,6 +488,51 @@ func Test_cumulativeCheckFees(t *testing.T) {
...
+								TxID:    fixture.AncestorTx1.TxID,
+								Bytes:   fixture.AncestorTx1.Bytes,
+								IsMined: false,
+							}
+							return []validation.RawTx{a1, fixture.AncestorTx2}, nil
+						}
+
+						if i == 2 {
+							return []validation.RawTx{fixture.AncestorOfAncestorTx1, fixture.AncestorOfAncestorTx2}, nil
+						}
```
Comment (boecklim) [inline internal/validator/default/default_validator_test.go:527]: IMO a switch would be easier to read

```
switch i {
case 0: ...
case 1: ...
case 2: ...
}
```

File: pkg/api/arc.yaml [hunk 2]
```diff
@@ -1078,6 +1080,13 @@ components:
         Whether we should perform cumulative fee validation for fee consolidation txs or not.
       schema:
         type: boolean
+    strictCumulativeFeeValidation:
+      name: X-StrictCumulativeFeeValidation
+      in: header
+      description: >-
+        Whether we should perform cumulative fee validation for fee consolidation txs or not with unmined ancestors limit.
```
Comment (boecklim) [inline pkg/api/arc.yaml:1087]: Whether we should perform cumulative fee validation for fee consolidation txs with unmined ancestors limit or not.

File: doc/README.md [hunk 3]
```diff
@@ -584,3 +584,31 @@ The system performs the fee validation and returns a result indicating that the
 - The `X-SkipFeeValidation` header takes precedence over `X-CumulativeFeeValidation` and causes the fee validation to be skipped.
 
 This feature is crucial to ensure that the chain of unmined transactions has sufficient fees, which is essential for the effective management of the transaction network.
+
+## Strict cumulative fees validation
+
+The fature is extention of the "Cumulative Fee Validation". It adds checking the lenght of unmined ancestors chain to validation.
```
Comment (boecklim) [inline doc/README.md:590]: There are some type-os in this line.

"This **feature** is **an** exten**s**ion of the "Cumulative Fee Validation". It adds checking the leng**th** of unmined ancestors chain to validation."


We should add a bit more explanation to that. So I would suggest this
"It adds checking the length of unmined ancestors chain against the setting of child pays for parent limitation `limitcpfpgroupmemberscount` of the connected nodes."

Comment (arkadiuszos4chain) [reply doc/README.md]: Changed

File: pkg/api/arc.yaml (hunk 2 above)
Comment (arkadiuszos4chain) [reply pkg/api/arc.yaml]: Changed

File: internal/validator/default/default_validator_test.go (hunk 1 above)
Comment (arkadiuszos4chain) [reply internal/validator/default/default_validator_test.go]: Changed

**Review by boecklim — APPROVED** [review]

**Review by kuba-4chain — APPROVED** [review]

**Comment by boecklim** [comment]
Replaced by https://github.com/bitcoin-sv/arc/pull/574.
PR for feature of `Strict Cumulative Fees Validation` may be re-opened later if needed.

**Pull request closed without merging**